### PR TITLE
Running the server without vscode - and optionally through stdio instead of sockets

### PR DESCRIPTION
**Pull request by ashfordneil**
I've changed the server so it now defaults to communicating over stdin / stdout instead of a socket if a port was not specified when launching. It currently comes up with a logging message stating that the socket has closed, which I still need to fix, but aside from that it appears functional to me.

I also added instructions to run the server manually to the readme.

Let me know what you think! :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
 [Install from the VS Code marketplace](https://marketplace.visualstudio.com/items?itemName=georgewfraser.vscode-javac)
 
+## Installation and Running for Other Editors (manually)
+
+- clone the repository
+- run `mvn package`
+- copy the produced jar file at `out/fat-jar.jar` to an appropriate location
+- run on a socket with `java -cp -Djavacs.port=${PORT} /path/to/fat-jar.jar org.javacs.Main`
+- run through stdin/stdout with `java -cp /path/to/fat-jar.jar org.javacs.Main`
```
Comment (Limeth) [inline README.md]: I believe that running just `java -jar /path/to/fat-jar.jar` works too.

**Comment by Limeth** [comment]
Seems to be working just fine, thanks!

**Comment by Limeth** [comment]
@ashfordneil By the way, have you had any luck getting some kind of auto-import working?

**Comment by georgewfraser** [comment]
Sorry I haven't been a very good maintainer lately...I finally got around to taking a look at this.

There isn't really any reason to support both socket and stdio, so I changed it to just support stdio in https://github.com/georgewfraser/vscode-javac/commit/29504ed7bb0d5d370d1baaec479be76ad32b1816

**Pull request closed without merging**